extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -220,6 +289,76 @@ export class AmpSelector extends AMP.BaseElement {
   }
 
   /**
+   * Handles click events for the selectables.
+   * @param {!Event} event
+   */
+  clickHandler_(event) {
+    let el = dev().assertElement(event.target);
+    if (!el) {
+      return;
+    }
+    if (!el.hasAttribute('option')) {
+      el = closest(el, e => e.hasAttribute('option'), this.element);
+    }
+    if (el) {
+      this.onOptionSelected_(el);
+    }
+  }
+
+  /**
+   * Handles keyboard events for the selectables.
+   * @param {!Event} event
+   */
+  keyDownHandler_(event) {
+    // Make currently selected option unfocusable
+    this.options_[this.focusedIndex_].tabIndex = -1;
```

Comment:
This should probably be done after determining the direction to go in (if any direction at all!)